From b231fbdb5c13ab402519c9bc21d19b45708d1bae Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Tue, 27 Oct 2020 13:57:43 -0700
Subject: [PATCH] Port 'make rearguard_tarballs' to Solaris 10

* Makefile (tarballs, etc.): Export AWK to submake, for Solaris 10
which needs something like 'make AWK=nawk rearguard_tarballs'.
(.KEEP_STATE): Remove.  .KEEP_STATE is a Solaris 'make' feature
that breaks 'make rearguard_tarballs' because it causes submakes
to mistakenly consider files to be out-of-date.  As .KEEP_STATE
is not specified by POSIX and is ignored by GNU 'make' it should
not be needed.
---
 Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index a0b6aeb..5064a19 100644
--- a/Makefile
+++ b/Makefile
@@ -1012,7 +1012,7 @@ tarballs rearguard_tarballs traditional_tarballs \
 signatures rearguard_signatures traditional_signatures: \
   version set-timestamps.out rearguard.zi
 		VERSION=`cat version` && \
-		$(MAKE) VERSION="$$VERSION" $@_version
+		$(MAKE) AWK='$(AWK)' VERSION="$$VERSION" $@_version
 
 # These *_version rules are intended for use if VERSION is set by some
 # other means.  Ordinarily these rules are used only by the above
@@ -1110,8 +1110,6 @@ strftime.o:	private.h tzfile.h
 zdump.o:	version.h
 zic.o:		private.h tzfile.h version.h
 
-.KEEP_STATE:
-
 .PHONY: ALL INSTALL all
 .PHONY: check check_time_t_alternatives
 .PHONY: check_web check_zishrink
-- 
1.8.3.1

